src/integrationTest/java/org/opensearch/security/DoNotFailOnForbiddenTests.java

```diff
@@ -0,0 +1,348 @@
+
+			assertThatThrownBy(() -> restHighLevelClient.search(searchRequest, DEFAULT), statusException(FORBIDDEN));
+		}
+	}
+
+	@Test
+	public void shouldSearchForDocumentsViaAlias_positive() throws IOException {
+		try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(LIMITED_USER)) {
+			SearchRequest searchRequest = queryStringQueryRequest(BOTH_INDEX_ALIAS, QUERY_TITLE_MAGNUM_OPUS);
+
+			SearchResponse searchResponse = restHighLevelClient.search(searchRequest, DEFAULT);
+
+			assertThat(searchResponse, isSuccessfulSearchResponse());
+			assertThat(searchResponse, numberOfTotalHitsIsEqualTo(1));
+			assertThat(searchResponse, searchHitsContainDocumentWithId(0, MARVELOUS_SONGS, "1"));
+			assertThat(searchResponse, searchHitContainsFieldWithValue(0, FIELD_TITLE, TITLE_MAGNUM_OPUS));
+		}
+	}
+
+	@Test
+	public void shouldSearchForDocumentsViaAlias_negative() throws IOException {
+		try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(LIMITED_USER)) {
+			SearchRequest searchRequest = queryStringQueryRequest(FORBIDDEN_INDEX_ALIAS, QUERY_TITLE_POISON);
+
+			assertThatThrownBy(() -> restHighLevelClient.search(searchRequest, DEFAULT), statusException(FORBIDDEN));
+		}
+	}
+
+	@Test
+	public void shouldSearchForDocumentsViaAll_positive() throws IOException {
+		try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(LIMITED_USER)) {
+			SearchRequest searchRequest = queryStringQueryRequest("_all", QUERY_TITLE_MAGNUM_OPUS);
+
+			SearchResponse searchResponse = restHighLevelClient.search(searchRequest, DEFAULT);
+
+			assertThat(searchResponse, isSuccessfulSearchResponse());
+			assertThat(searchResponse, numberOfTotalHitsIsEqualTo(1));
+			assertThat(searchResponse, searchHitsContainDocumentWithId(0, MARVELOUS_SONGS, "1"));
+			assertThat(searchResponse, searchHitContainsFieldWithValue(0, FIELD_TITLE, TITLE_MAGNUM_OPUS));
+		}
+	}
+
+	@Test
+	public void shouldSearchForDocumentsViaAll_negative() throws IOException {
+		try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(LIMITED_USER)) {
+			SearchRequest searchRequest = queryStringQueryRequest("_all", QUERY_TITLE_POISON);
+
+			SearchResponse searchResponse = restHighLevelClient.search(searchRequest, DEFAULT);
+
+			assertThat(searchResponse, isSuccessfulSearchResponse());
+			assertThat(searchResponse, numberOfTotalHitsIsEqualTo(0));
+		}
+	}
+
+	@Test
+	public void shouldMSearchDocument_positive() throws IOException {
+		try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(LIMITED_USER)) {
+			MultiSearchRequest request = new MultiSearchRequest();
+			request.add(queryStringQueryRequest(BOTH_INDEX_PATTERN, QUERY_TITLE_MAGNUM_OPUS));
+			request.add(queryStringQueryRequest(BOTH_INDEX_PATTERN, QUERY_TITLE_NEXT_SONG));
+
+			MultiSearchResponse response = restHighLevelClient.msearch(request, DEFAULT);
+
+			assertThat(response, notNullValue());
```

Comment:
Nit: `notNullValue` isn't very useful if the values themselves are evaluated later in the tests, I'd recommend removing it to keep the tests easier to read

Reply:
done